tools/interu/src/config/profile.rs

```diff
@@ -0,0 +1,153 @@
+pub enum Strategy {
+    Weighted(WeightedOptions),
+    UseRunner(UseRunnerOptions),
+}
+
+impl Strategy {
+    pub fn validate(
+        &self,
+        profile_name: &str,
+        runners: &BTreeMap<String, Runner>,
+    ) -> Result<(), StrategyValidationError> {
+        match &self {
+            Strategy::Weighted(weighted_options) => {
+                weighted_options.validate(profile_name, runners)
+            }
+            Strategy::UseRunner(use_runner_options) => {
+                use_runner_options.validate(profile_name, runners)
+            }
+        }
+    }
+
+    pub fn get_test_options(&self) -> (usize, &TestRun, &str) {
+        match self {
+            Strategy::Weighted(options) => (
+                options.options.parallelism,
+                &options.options.test_run,
+                options.options.test_parameter.as_str(),
+            ),
+            Strategy::UseRunner(options) => (
+                options.options.parallelism,
+                &options.options.test_run,
+                options.options.test_parameter.as_str(),
+            ),
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct WeightedOptions {
+    pub weights: Vec<Weight>,
+    pub options: TestOptions,
+}
+
+impl WeightedOptions {
+    pub fn validate(
+        &self,
+        profile_name: &str,
+        runners: &BTreeMap<String, Runner>,
+    ) -> Result<(), StrategyValidationError> {
+        for weight in &self.weights {
+            if !runners.contains_key(&weight.runner) {
+                return InvalidRunnerReferenceSnafu {
+                    at: format!(
+                        "profile.{profile_name}.weights.{weight}",
+                        weight = weight.weight
+                    ),
+                    runner_ref: weight.runner.clone(),
+                }
+                .fail();
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct Weight {
+    pub weight: usize,
+    pub runner: String,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct UseRunnerOptions {
+    pub runner: String,
+    pub options: TestOptions,
+}
+
+impl UseRunnerOptions {
+    pub fn validate(
+        &self,
+        profile_name: &str,
+        runners: &BTreeMap<String, Runner>,
+    ) -> Result<(), StrategyValidationError> {
+        if !runners.contains_key(&self.runner) {
+            return InvalidRunnerReferenceSnafu {
+                at: format!("profile.{profile_name}.runner"),
+                runner_ref: self.runner.clone(),
+            }
+            .fail();
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case")]
+pub struct TestOptions {
+    pub parallelism: usize,
+
+    #[serde(default, with = "serde_yaml::with::singleton_map")]
```

Comment:
Not 100% sure I understand what singleton_map is doing here.
I read this: https://strawlab.org/strand-braid-api-docs/latest/serde_yaml/with/singleton_map/index.html

It appears to have no effect for variants that hold no data.
```

Reply:
Yes, that argument can be removed as it is currently not needed.

Reply:
Removed in 4077bf5